### ISSUE TITLE
[BUG] Provide valid swagger docs

### ISSUE THREAD

**Issue by drewmullen** [harvester/harvester#5211]
**Describe the bug**
Thank you for your work on this project. I am trying to generate a swagger client but I cannot because the document contains errors. The issues stem from an improper use of regex in the document. Instead you should use [swagger patterns](https://swagger.io/docs/specification/data-models/data-types/#pattern) (v3). Note: the current is invalid in both v2 and v3

For example:

[Current regex usage](https://raw.githubusercontent.com/harvester/harvester/master/api/openapi-spec/swagger.json)

```yaml
  paths:
    /api/v1/namespaces/{namespace:[a-z0-9][a-z0-9\\-]*}/persistentvolumeclaims: 
```

Should be
```yaml
paths: 
  /api/v1/namespaces/{namespace}/persistentvolumeclaims:
...
parameters:
- name: namespace
  in: path
  description: "Object name and auth scope, such as for teams and projects"
  required: true
  schema:
    pattern: '[a-z0-9][a-z0-9-]*'
    type: string
```

 

**To Reproduce**
```shell
$ swagger validate https://raw.githubusercontent.com/harvester/harvester/master/api/openapi-spec/swagger.json
```

**Confirming issues are fixed**

```shell
$ swagger generate client -f ./swagger.json -A harvester-go-sdk
```

I'm working on a PR that will also upgrade to oas v3

**Comment by drewmullen** [comment]
I'm also getting an error somewhere that the doc is missing a definition of `annotations`:
```shell
go run main.go                                                                                 
Error when calling `VirtualMachinesAPI.ListNamespacedVirtualMachine``: json: unknown field "annotations"
```

When i check the API it doesnt in fact return annotations (which is obvious but just pointing it out for completeness
<img width="457" alt="image" src="https://github.com/harvester/harvester/assets/3812162/1587edac-bba5-4265-8b73-56b46c60c130">

**Comment by drewmullen** [comment]
This was resolved by #5319
